### PR TITLE
Support new properties in posthog UTD reports

### PR DESCRIPTION
**Pull request by richvdh**
Add a few new properties to the UTD reports we send to Posthog.

Part of https://github.com/element-hq/element-meta/issues/2582

Requires https://github.com/matrix-org/matrix-rust-sdk/pull/4404.

Signed-off-by: Richard van der Hoff \<richardv@element.io>

## Tests

- Receive a UTD
- check the logging looks plausible

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): Android 15

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

**Comment by bmarty** [comment]
Hello @richvdh https://github.com/element-hq/element-x-android/pull/4024 has been merged, I think it contains what you need for this PR. You will have to update the tests and probably add some (see the modified change in https://github.com/element-hq/element-x-android/pull/4024). Let me know if you need anything!

**Review by bmarty — APPROVED** [review]
Thanks!

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/analytics/UtdTracker.kt [hunk 1]
```diff
@@ -39,6 +39,10 @@ class UtdTracker(
             timeToDecryptMillis = info.timeToDecryptMs?.toInt() ?: -1,
             domain = Error.Domain.E2EE,
             name = name,
+            eventLocalAgeMillis = info.eventLocalAgeMillis.toInt(),
+            userTrustsOwnIdentity = info.userTrustsOwnIdentity,
+            isFederated = info.ownHomeserver != info.senderHomeserver,
+            isMatrixDotOrg = info.ownHomeserver == "matrix.org",
```
[the remark this answers is not in the document]
Comment (richvdh) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/analytics/UtdTracker.kt:45]: I don't quite follow. The name of the `matrix.org` homeserver is `matrix.org`.

This matches the behaviour on Element Web: https://github.com/element-hq/element-web/blob/edaf9773c02e71b20eedebbd9619ff8a2b2e4cf2/src/DecryptionFailureTracker.ts#L322.

[the remark this answers is not in the document]
Comment (bmarty) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/analytics/UtdTracker.kt:45]: ok, thanks!

**Pull request merged**